mailpoet/lib/API/JSON/v1/AutomatedLatestContent.php

```diff
@@ -65,14 +75,24 @@ public function getTerms($data = []) {
     return $this->successResponse(array_values($terms));
   }
 
+  /**
+   * @param int[]|\WP_Post[] $posts
+   * @return \WP_Post[]
+   */
+  private function getPermittedPosts($posts) {
+    return array_filter($posts, function ($post) {
+      return !is_numeric($post) && $this->permissionHelper->checkReadPermission($post);
```

Comment:
Why `!is_numeric()`? I mean, why expect numeric values at all? I know `get_posts()` can return `int[]` with `["fields" => "ids"]` which we don't use though. Maybe we could make sure, this does not happen for posts returned from ACL, so that we do not have this ambiguity trickling down in our code base?

Reply:
@websupporter since we were are not preventing the int[] output, this is needed per type definitions! not sure if we want to prevent this behaviour now, I can create a Jira ticket to check this in depth and probably refactor, what do you think?